src/filefinder.cpp

```diff
@@ -795,8 +795,10 @@ FileFinder::Directory FileFinder::GetDirectoryMembers(const std::string& path, F
 			#ifdef PSP2
 			is_directory = S_ISDIR(ent.d_stat.st_mode);
 			#else
+			#ifndef __HAIKU__
 			is_directory = ent->d_type == DT_DIR;
```

Comment:
I don't think this is a good solution because it keeps "is_directory" not initialized.
But this line really needs improvement in general, DT_UNKNOWN isn't properly handled, either...

Reply:
Hi @Ghabry. Very true, we are missing something here, I'll get a patch sent asap.

Regarding DT_UNKNOWN, it's a bit more complicated as d_type is not POSIX and there is no equivalent in Haiku (or Solaris) and will require more rewriting I'm afraid. Here's what I can work with in stat.h : 

```
/* standard file types */
#define S_IFMT                          00000170000 /* type of file */
#define S_IFSOCK                        00000140000 /* socket */
#define S_IFLNK                         00000120000 /* symbolic link */
#define S_IFREG                         00000100000 /* regular */
#define S_IFBLK                         00000060000 /* block special */
#define S_IFDIR                         00000040000 /* directory */
#define S_IFCHR                         00000020000 /* character special */
#define S_IFIFO                         00000010000 /* fifo */

#define S_ISREG(mode)           (((mode) & S_IFMT) == S_IFREG)
#define S_ISLNK(mode)           (((mode) & S_IFMT) == S_IFLNK)
#define S_ISBLK(mode)           (((mode) & S_IFMT) == S_IFBLK)
#define S_ISDIR(mode)           (((mode) & S_IFMT) == S_IFDIR)
#define S_ISCHR(mode)           (((mode) & S_IFMT) == S_IFCHR)
#define S_ISFIFO(mode)          (((mode) & S_IFMT) == S_IFIFO)
#define S_ISSOCK(mode)          (((mode) & S_IFMT) == S_IFSOCK)
#define S_ISINDEX(mode)         (((mode) & S_INDEX_DIR) == S_INDEX_DIR)

#define S_IUMSK 07777           /* user settable bits */

#define S_ISUID 04000       /* set user id on execution */
#define S_ISGID 02000       /* set group id on execution */

#define S_ISVTX 01000       /* save swapped text even after use (sticky bit) */

#define S_IRWXU 00700       /* read, write, execute: owner */
#define S_IRUSR 00400       /* read permission: owner */
#define S_IWUSR 00200       /* write permission: owner */
#define S_IXUSR 00100       /* execute permission: owner */
#define S_IRWXG 00070       /* read, write, execute: group */
#define S_IRGRP 00040       /* read permission: group */
#define S_IWGRP 00020       /* write permission: group */
#define S_IXGRP 00010       /* execute permission: group */
#define S_IRWXO 00007       /* read, write, execute: other */
#define S_IROTH 00004       /* read permission: other */
#define S_IWOTH 00002       /* write permission: other */
#define S_IXOTH 00001       /* execute permission: other */

/* BSD extensions */
#define S_IREAD         S_IRUSR
#define S_IWRITE        S_IWUSR
#define S_IEXEC         S_IXUSR
```

Reply:
Took care of the init in https://github.com/libretro/easyrpg-libretro/pull/14